test/dlc_tests/conftest.py

```diff
@@ -376,6 +381,10 @@ def framework_version_within_limit(metafunc_obj, image):
         pt14_requirement_failed = "pt14_and_above_only" in metafunc_obj.fixturenames and is_below_framework_version("1.4", image, "pytorch")
         if pt16_requirement_failed or pt15_requirement_failed or pt14_requirement_failed:
             return False
+    if image_framework_name.startswith("huggingface_"):
+        non_huggingface_requirement_failed = "non_huggingface_only" in metafunc_obj.fixturenames
+        if non_huggingface_requirement_failed:
+            return False
```

Comment:
Please move this to the part of the code where `cpu_only`, `gpu_only`, `py3_only`, etc. are evaluated. This function is specifically dedicated to framework versions.